ddl/db_test.go

```diff
@@ -1711,6 +1711,54 @@ func (s *testDBSuite) TestCreateTableWithPartition(c *C) {
 		  partition p3 values less than (18446744073709551000 + 1),
 		  partition p4 values less than (18446744073709551000 + 10)
 		);`)
+
+	// Support partitioned by range columns later "ERROR HY000: Field 'a' is of a not allowed type for this type of partitioning".
+	s.tk.MustExec(`create table t29 (
+		a decimal
+	)
+	partition by range columns (a)
+	(partition p0 values less than (0));`)
+}
+
+func (s *testDBSuite) TestCreateTableWithHashPartition(c *C) {
```

Comment:
What is the purpose of this test case?

Reply:
Check grammar compatibility for different table partition type? @shenli